tools/kotlin/BUILD.bazel

```diff
@@ -0,0 +1,32 @@
+"""
+Configurations for a codebase-wide build toolchain for Kotlin.
+"""
+
+load("@io_bazel_rules_kotlin//kotlin:core.bzl", "define_kt_toolchain", "kt_javac_options", "kt_kotlinc_options")
+
+# This exposes a patch file that retroactively adds opt-in support to kt_kotlinc_options (which was
+# added in a later version of rules_kotlin that's not available for the version of Bazel used at the
+# time of this introduction).
+exports_files(["add_kotlinc_optin_support.patch"])
+
+kt_kotlinc_options(
+    name = "oppia_kotlinc_options",
+    warn = "error",
```

Comment:
Reminder to update this to be 'report' for local builds and 'error' for CI builds.

Reply:
Is this to be done as part of this PR?

Reply:
Yep, will be included in the latest batch of changes.